altair/utils/core.py

```diff
@@ -532,6 +532,23 @@ def parse_shorthand(
             if isinstance(attrs["type"], tuple):
                 attrs["sort"] = attrs["type"][1]
                 attrs["type"] = attrs["type"][0]
+
+    # If an unescaped colon is still present, it's often due to an incorrect data type specification
+    # but could also be due to using a column name with ":" in it.
+    if (
+        "field" in attrs
+        and ":" in attrs["field"]
+        and attrs["field"][attrs["field"].rfind(":") - 1] != "\\"
+    ):
+        raise ValueError(
+            '"{}" '.format(attrs["field"].split(":")[-1])
+            + "is not one of the valid encoding data types: {}.".format(
+                ", ".join(TYPECODE_MAP.values())
+            )
+            + "\nFor more details, see https://altair-viz.github.io/altair-docs/user_guide/encodings/index.html#encoding-data-types. "
+            + "If you are trying to use a column name that contains a colon, "
+            + 'prefix it with a backslash; for example "column\\:name" instead of "column:name".'
```

Comment:
```suggestion
            + 'prefix it with a backslash; for example "column\:name" instead of "column:name".'
```
singe `\` colon isn't it?

Reply:
The additional backslash is needed to make sure a single backslash is shown in the outputted warning message (the first backslash makes sure the second one is printed instead of being interpreted as an escape character for the following colon):

![image](https://user-images.githubusercontent.com/4560057/220473592-1552fb76-f5f2-4ba2-87c0-68ae25dd37d0.png)